test/jasmine/tests/axes_test.js

```diff
@@ -1858,24 +1858,21 @@ describe('Test axes', function() {
             expect(ax._max).toEqual([{val: 6, pad: 10, extrapad: true}]);
         });
 
-        it('should return early if no data is given', function() {
+        it('should fail if no data is given', function() {
```

Comment:
you're confident we always send an array to `expand`? I guess in principle if we don't have a required array the trace will be set `visible: false`...

Reply:
Yes, `expand` is only called from trace module `calc` or `setPosition` or component `calcAutorange` which are skipped for `visible: false` traces / component items.

I thought removing that `if(!data) return;` was a nice way to `expand` is only called when needed.